### PR TITLE
RDM-7423 RDM-7424 Adjust DSL elements

### PR DESCRIPTION
**Pull request by rafalkalita**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-7424
https://tools.hmcts.net/jira/browse/RDM-7423

### Change description ###
Extra Space Tolerance Between Words of DSL Elements
Optional Comma or Full Stop At End Of DSL Elements

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Comment by mattnayler** [comment]
The draft PR hmcts/ccd-data-store-api#805 has been created to demonstrate these changes working against this branch: by using release tag:- [1.0.22.RDM-7423-7424](/hmcts/ccd-api-test-automation-fw/releases/tag/1.0.22.RDM-7423-7424).  

> The tests in feature file `F-026` have been updated to simulate the DSL changes covered by [RDM-7423](https://tools.hmcts.net/jira/browse/RDM-7423) and [RDM-7424](https://tools.hmcts.net/jira/browse/RDM-7424).

File: src/main/java/uk/gov/hmcts/befta/util/CucumberStepAnnotationUtils.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+public final class CucumberStepAnnotationUtils {
+
+    // Hide Utility Class Constructor : Utility classes should not have a public or default constructor (squid:S1118)
+    private CucumberStepAnnotationUtils() { }
+
+    public static void adjustCucumberStepAnnotations(Class<?> testClass) {
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/befta/util/CucumberStepAnnotationUtils.java]: please rename this to --> injectCommonSyntaticFlexibilitiesIntoStepDefinitionsIn
This name will tell more directly what the method is doing.

Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/befta/util/CucumberStepAnnotationUtils.java]: The function name has now been updated to: `injectCommonSyntacticFlexibilitiesIntoStepDefinitions`

**Comment by mattnayler** [comment]
The draft test PR hmcts/ccd-data-store-api#805 has been updated to demonstrate these changes working against this updated branch: by using release tag:- [1.0.22.RDM-7423-7424.2](/hmcts/ccd-api-test-automation-fw/releases/tag/1.0.22.RDM-7423-7424.2).


NB: two test are failing in the test PR (hmcts/ccd-data-store-api#805) however these tests are currently also failing for other PRs in this branch at this current time. 

__Failed scenarios:__
* `src/aat/resources/features/F-026/F-026.feature:8` # [RDM-7423: Optional Comma] must return the list of cases and status code 200 for correct inputs
  * `Transfer-Encoding is unavailable though it was expected to be there`
* `src/aat/resources/features/F-051/F-051.feature:8` # must return default user setting successfully for a user having a profile in CCD
  * `Transfer-Encoding is unavailable though it was expected to be there`

**Pull request merged**